File: java/SimpleAk/src/main/java/mol/SetRequest.java
package mol;

import java.io.Serializable;

/**
 * Created by cqshinn on 2/11/16.
 */
public class SetRequest implements Serializable {
    private final String key;
    private final Object value;

    public SetRequest(String key, Object value) {
        this.key = key;
        this.value = value;
    }

    public String getKey() {
        return key;
    }

    public Object getValue() {
        return value;
    }
}

File: scala/SimpleAk/src/main/scala-2.11/mol/SetRequest.scala
package mol

/**
 * Created by cqshinn on 2/11/16.
 */
case class SetRequest(val key: Option[String], val value: Option[String])
case class GetRequest(val key: Option[String]) extends Serializable
case class KeyNotFoundException(val key: Option[String]) extends Exception with Serializable

